### DISCUSSION TITLE
Channel and point bar modelling

### DISCUSSION THREAD

**Discussion by KirillSimonov** [gempy-project/gempy#589] [Ideas]
Dear GemPy team,

Thanks for such a beautiful open-source tool!
I am beginner modeler and now I want to create channel, abandoned channel and point bar structures and secondly do the forward seismic modelling?

Could you please tell if this is applicable using GemPy as I did not find any information on this type of modelling?

Thanks.

---
Sincerely,
Kirill

**Comment by javoha** [comment]
Hello @KirillSimonov ,

generally speaking GemPy works best for settings with parallel to subparallel sets ("series") of layers, due to the implicit nature of the underlying algorithm. We are working on options to allow explicit modelling of channel systems and similar "objects". As of now it really depends ont the scale and accuracy of your problem and required model. If you can provide some sort of rough sketch of your expected geologoy we might get a better idea if that would be feasible at the moment.

Hopefully this is helpful as a starting point, stay healthy and have a nice day!

**Comment by KirillSimonov** [comment]
Hello @javoha ,

Well, parallel strata will be fine for me as then I process it as 2D image, so this is not a problem. 
I attach example images of ideal cases with channels, and channels plus point bars to be modelled. After modelliing I want to do forward seismic modelling to create seismic data. Also, I attach example image of real seismic data that I want to mimic. Please see below.

Thank you! Stay safe and have a nice day!

![2020-05-19_11h30_45](https://user-images.githubusercontent.com/45141680/82304037-755eac80-99c4-11ea-9d7a-a645f14f9235.png)
![2020-05-19_11h24_46](https://user-images.githubusercontent.com/45141680/82304044-77c10680-99c4-11ea-86be-ef6bc841e7ac.png)

**Comment by KirillSimonov** [comment]
Hello @javoha ,
Are there any news on my previous post?

**Comment by flohorovicic** [comment]
Actually, looks like something for @MohammadCGRE - did you have a look, yet? Could be better to model these structures with an object-based approach or an explicit representation.

**Comment by MohammadCGRE** [comment]
Hello @KirillSimonov 

First of all Thanks @javoha and @flohorovicic. Sorry for my late answer (I just saw this message today). Exactly like what @javoha and @flohorovicic said, you can use explicit modelling for these channels by GempyExplicit. You will have some limited control points for managing the structures. GempyExplicit is now ready but it is in the experimental version. It will be accessible to the public in the near future.
![image (1)](https://user-images.githubusercontent.com/61519597/84017149-8aed4380-a97e-11ea-8fb7-112f1ef5fd58.png)

**Comment by KirillSimonov** [comment]
Dear @flohorovicic and @MohammadCGRE ,
 Thank you! This is what I want. I saw similar functionality in Petrel, but there are rooms to explore.
 Is GempyExplicit separate package from GemPy or it is inner library and will available soon?

**Comment by MohammadCGRE** [comment]
Dear @KirillSimonov 

GempyExplicit is a separate package of gempy. But you can import both of them to your project. Right now gempyExplicit is in the experimental version but probably I can share it with you in the near future...if you want..to use it.
 Do you want to make  Just channels?

**Comment by KirillSimonov** [comment]
Dear @MohammadCGRE ,
Sure, I would like to try it out if it is possible. I want to generate channels, aggraded channels and point bars. What are other options for geological bodies?
Thank you

**Comment by KirillSimonov** [comment]
Dear @MohammadCGRE ,
Could you please tell  if GempyExplicit is available for testing?
 
Thanks.

**Comment by MohammadCGRE** [comment]
Dear @KirillSimonov 

Hello, Thank you very much for your interest. We are writing a paper based on the gempyExplicit and after publishing the paper we are allowed to release it. I will let you know when it is ready to use.
Best,
Mohammad.

**Comment by KirillSimonov** [comment]
Dear @MohammadCGRE ,

Looking forward to try it out!
Thank you.
